haas/api.py

```diff
@@ -428,49 +428,36 @@ def node_detach_network(node, nic, network):
 
 
 @rest_call('PUT', '/node/<node>/metadata/<label>', Schema({
-    'node': basestring, 'label': basestring, 'value': basestring
+    'node': basestring, 'label': basestring, 'value': object,
 }))
-def node_register_metadata(node, label, value):
+def node_set_metadata(node, label, value):
     """Register metadata on a node.
 
-    If the label already exists, a DuplicateError will be raised.
+    If the label already exists, the value will be updated.
     """
     get_auth_backend().require_admin()
     node = _must_find(model.Node, node)
-    _assert_absent_n(node, model.Metadata, label)
-    metadata = model.Metadata(label, value, node)
-
-    db.session.add(metadata)
-    db.session.commit()
-
-
-@rest_call('PUT', '/node/<node>/update/metadata/<label>', Schema({
-    'node': basestring, 'label': basestring, 'value': basestring
-}))
-def node_update_metadata(node, label, value):
-    """update metadata on a node.
-
-    If the metadata does not exist, a NotFoundError will be raised.
-    """
-    get_auth_backend().require_admin()
-    node = _must_find(model.Node, node)
-    metadata = _must_find_n(node, model.Metadata, label)
-    metadata.value = value
-
+    obj_inner = _namespaced_query(node, model.Metadata, label)
+    if obj_inner is not None:        
+        metadata = _must_find_n(node, model.Metadata, label)
+        metadata.value = value
```

Comment:
You'll need to convert `value` to a string to store it in the db (and do the reverse when you pull it out). Just a call to `json.dumps` will do.